cooltools/dotfinder.py

```diff
@@ -1065,22 +1071,56 @@ def scoring_step(clr, expected, expected_name, tiles, kernels,
         # consider using
         # https://github.com/mirnylab/cooler/blob/9e72ee202b0ac6f9d93fd2444d6f94c524962769/cooler/tools.py#L59
         # here:
+        chunks = map_(job, tiles, **map_kwargs)
         ###########################################
         #
         # this is to be rewritten using cooler output
         # as in https://github.com/mirnylab/cooltools/issues/51#issuecomment-458664781
         #
         ###########################################
-        chunks = map_(job, tiles, **map_kwargs)
-        append = False
-        for chunk in chunks:
-            chunk.to_hdf(output_path,
-                         key='results',
-                         format='table',
-                         complevel=9,
-                         complib="blosc:snappy",
-                         append=append)
-            append = True
+        if output_mode == "hdf":
+            print("pandas hdf output ...")
+            append = False
+            for chunk in chunks:
+                chunk.to_hdf(output_path,
+                             key='results',
+                             format='table',
+                             complevel=9,
+                             complib="blosc:snappy",
+                             append=append)
+                append = True
+        # ###########################################
+        # #
+        # # cooler as an output for the dump:
+        # #
+        # ###########################################
+        if output_mode == "cooler":
+            print("cooler output ...")
+            columns = ["la_exp."+k+".value" for k in kernels]
+            # extra columns would be float64 by default ...
+            # dtypes = {'la_exp.donut.value': np.float64, ...}
+            cooler.create_cooler(output_path,
+                                clr.bins()[:],
+                                chunks, # iterator of pixel chunks ...
+                                columns=columns,
+                                # # should be float64 by default:
+                                # dtypes=dtypes,
+                                # to be included later ?!
+                                # # metadata : dict, optional
+                                # # assembly : str, optional
+                                ordered = True, # is it really ? should be
```

Comment:
doesn't seem like ordered at all ........
trying to figure out .....
probably found some api description inconsistency in cooler https://github.com/mirnylab/cooler/issues/149